stl.go

```diff
@@ -215,20 +225,46 @@ func ReadFromSTL(i io.Reader) (o *Subtitles, err error) {
 			continue
 		}
 
+		var starttime = g.timecodeStartOfProgramme
+		if starttime > t.timecodeIn {
+			starttime = 0
+		}
 		// Create item
 		var i = &Item{
-			EndAt:   t.timecodeOut - g.timecodeStartOfProgramme,
-			StartAt: t.timecodeIn - g.timecodeStartOfProgramme,
+			EndAt:    t.timecodeOut - starttime,
+			Metadata: &ItemMetadata{},
+			StartAt:  t.timecodeIn - starttime,
 		}
 
 		// Loop through rows
 		for _, text := range bytes.Split(t.text, []byte{0x8a}) {
 			parseTeletextRow(i, ch, func() styler { return newSTLStyler() }, text)
 		}
 
-		// Append item
-		o.Items = append(o.Items, i)
+		if i.InlineStyle == nil {
+			i.InlineStyle = &StyleAttributes{}
+		}
+
+		i.InlineStyle.STLVerticalPostion = newSTLVerticalPosition(t.verticalPosition)
+		i.InlineStyle.STLJustificationCode = newSTLJustificationCode(t.justificationCode)
 
+		i.InlineStyle.propagateSTLAttributes()
+
+		//Get ItemMetadata
+		i.Metadata = &ItemMetadata{
+			Index: t.subtitleNumber,
+		}
+
+		i.Metadata.STLCommentFlag = &t.commentFlag
+		i.Metadata.STLCumulativeStatus = &t.cumulativeStatus
+		i.Metadata.STLExtensionBlockNumber = &t.extensionBlockNumber
+		i.Metadata.STLSubtitleGroupNumber = &t.subtitleGroupNumber
+		i.Metadata.STLText = &t.text
```

Comment:
I don't like storing the complete text here (it means that this lib is not capable of parsing colors inside this text) but this will do for now

Reply:
Apart from colours, spaces in the text also ara important for us

Reply:
If the text is from STL to STL, we dont need to encode. We need to mantain spaces, colour, height, font style that comes in text field